consensus/state_processing/src/per_block_processing/process_operations.rs

```diff
@@ -127,6 +125,8 @@ pub mod altair_deneb {
         state.build_committee_cache(RelativeEpoch::Current, spec)?;
 
```

Comment:
We use the `epoch_cache` and `slashings_cache` in this function but I'm not sure how we ensure they've already been initialized (and initialized to the correct epoch) by this point.

Also, regardless of whether we add the guards to initialize them here, should we move the calls to `state.build_committee_cache()` up one level to `process_attestations()` since this function is called in a loop?

Reply:
They're initialised by top-level block processing here:

https://github.com/sigp/lighthouse/blob/efb457bfb4aa96e16682ffbb98ec2c87d6e9b2c2/consensus/state_processing/src/per_block_processing.rs#L122-L125

Although it wouldn't hurt to check the initialisation again at the start of attestation processing. I'll make that change